src/compiler/crystal/compiler.cr

```diff
@@ -580,7 +612,17 @@ module Crystal
 
     private def optimize_with_new_pass_manager(llvm_mod)
       LLVM::PassBuilderOptions.new do |options|
-        LLVM.run_passes(llvm_mod, "default<O3>", target_machine, options)
+        s = case @optimization_mode
+            when OptimizationMode::O3
+              "default<O3>"
+            when OptimizationMode::O2
+              "default<O2>"
+            when OptimizationMode::O1
+              "default<O1>"
+            else
+              "default<O0>"
+            end
```

Comment:
**suggestion:** Again, I'd recommend exhaustive case with predicate methods. Building a string as `"default<#{@optimization_mode}>"` could work as well, but since these are external option values, it's probably better to be explicit.
```suggestion
        s = case @optimization_mode
            in .o3? then "default<O3>"
            in .o2? then "default<O2>"
            in .o1? then "default<O1>"
            in .o0? then "default<O0>"
            end
```